### PR TITLE
Documentation for custom field selects

### PR DESCRIPTION
**Pull request by forsner**
This PR updates the documentation for Custom Fields and adds documentation for Custom Field Selects and Options.

This PR is part of a fix for https://github.com/Teamtailor/teamtailor/issues/10277

### CODE REVIEW

File: source/includes/custom-field-options/_create.md.erb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## Create Custom Field Option
+
+```http
+POST https://api.teamtailor.com/v1/custom-fields HTTP/1.1
```
Comment (emilkarl) [inline source/includes/custom-field-options/_create.md.erb:4]: Is this the correct url?

File: source/includes/custom-field-options/_create.md.erb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          "type": "custom-fields"
+        }
+      }
+		}
+  }
+}
+```
+
+```shell
+curl -X "POST" "https://api.teamtailor.com/v1/custom-fields" \
```
Comment (emilkarl) [inline source/includes/custom-field-options/_create.md.erb:27]: Is this the correct url?

File: source/includes/custom-field-options/_update.md.erb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+```shell
+curl -X "PATCH" "https://api.teamtailor.com/v1/custom-field-values/1" \
+     -H "Authorization: Token token=abc123abc123" \
+     -H "X-Api-Version: <%= config[:api_version] %>"
+     -H "Content-Type: application/vnd.api+json" \
+     -d '{
+   "data": {
+     "id": 1,
+     "type": "custom-field-options",
```
Comment (emilkarl) [inline source/includes/custom-field-options/_update.md.erb:27]: Is this the correct type, shouldn't it be `custom-field-values`?

File: source/includes/custom-field-options/_update.md.erb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+```http
+PATCH https://api.teamtailor.com/v1/custom-field-values/1 HTTP/1.1
+Authorization: Token token=abc123abc123
+X-Api-Version: <%= config[:api_version] %>
+
+{
+  "data": {
+    "id": 1,
+    "type": "custom-field-options",
```
Comment (emilkarl) [inline source/includes/custom-field-options/_update.md.erb:11]: Is this the correct type, shouldn't it be `custom-field-values`?

File: source/includes/custom-field-selects/_create.md.erb [hunk 5]
```diff
@@ -0,0 +1,87 @@
+## Create Custom Field Selects
+
+```http
+POST https://api.teamtailor.com/v1/custom-fields HTTP/1.1
```
Comment (emilkarl) [inline source/includes/custom-field-selects/_create.md.erb:4]: Is this the correct url?

File: source/includes/custom-field-options/_create.md.erb (hunk 1 above)
Comment (forsner) [reply source/includes/custom-field-options/_create.md.erb:4]: no 🙁

File: source/includes/custom-field-selects/_update.md.erb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+## Update Custom Field Selects
+
+```http
+PATCH https://api.teamtailor.com/v1/custom-field-values/1 HTTP/1.1
```
Comment (emilkarl) [inline source/includes/custom-field-selects/_update.md.erb:4]: Is this the correct url, i would expect `custom-field-selects` here?

**Review by emilkarl — COMMENTED** [review]
I few comments on the different urls?

File: source/includes/custom-field-options/_update.md.erb (hunk 3 above)
Comment (forsner) [reply source/includes/custom-field-options/_update.md.erb:27]: no, this is correct. It updates the custom field option value. not the selected value from custom field.

File: source/includes/custom-field-selects/_update.md.erb (hunk 6 above)
Comment (forsner) [reply source/includes/custom-field-selects/_update.md.erb:4]: you are correct.

File: source/includes/custom-field-options/_update.md.erb (hunk 4 above)
Comment (forsner) [reply source/includes/custom-field-options/_update.md.erb:11]: correct, same as above.

**Comment by forsner** [comment]
> I few comments on the different urls?

They are fixed now. Thank you 😊

File: source/includes/custom-field-options/_update.md.erb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      "self": "https://api.teamtailor.localhost/v1/custom-field-options/1"
+    },
+    "attributes": {
+      "value": "Updated value"
+    },
+    "relationships": {
+      "custom-field": {
+        "links": {
+          "self": "https://api.teamtailor.localhost/v1/custom-field-options/1/relationships/custom-field",
+          "related": "https://api.teamtailor.localhost/v1/custom-field-options/1/custom-field"
```
Comment (frysch) [inline source/includes/custom-field-options/_update.md.erb:52]:
```suggestion
      "self": "https://api.teamtailor.com/v1/custom-field-options/1"
    },
    "attributes": {
      "value": "Updated value"
    },
    "relationships": {
      "custom-field": {
        "links": {
          "self": "https://api.teamtailor.com/v1/custom-field-options/1/relationships/custom-field",
          "related": "https://api.teamtailor.com/v1/custom-field-options/1/custom-field"
```

File: source/includes/custom-field-options/_create.md.erb [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      "self": "https://api.teamtailor.localhost/v1/custom-field-options/1"
+    },
+    "attributes": {
+      "value": "Option value"
+    },
+    "relationships": {
+      "custom-field": {
+        "links": {
+          "self": "https://api.teamtailor.localhost/v1/custom-field-options/1/relationships/custom-field",
+          "related": "https://api.teamtailor.localhost/v1/custom-field-options/1/custom-field"
```
Comment (frysch) [inline source/includes/custom-field-options/_create.md.erb:68]:
```suggestion
      "self": "https://api.teamtailor.com/v1/custom-field-options/1"
    },
    "attributes": {
      "value": "Option value"
    },
    "relationships": {
      "custom-field": {
        "links": {
          "self": "https://api.teamtailor.com/v1/custom-field-options/1/relationships/custom-field",
          "related": "https://api.teamtailor.com/v1/custom-field-options/1/custom-field"
```

File: source/includes/custom-field-options/_index.md.erb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+          "self": "https://api.teamtailor.localhost/v1/custom-field-options/1"
+        },
+        "attributes": {
+          "value": "Option 1"
+        },
+        "relationships": {
+          "custom-field": {
+            "links": {
+              "self": "https://api.teamtailor.localhost/v1/custom-field-options/1/relationships/custom-field",
+              "related": "https://api.teamtailor.localhost/v1/custom-field-options/1/custom-field"
```
Comment (frysch) [inline source/includes/custom-field-options/_index.md.erb:26]:
```suggestion
          "self": "https://api.teamtailor.com/v1/custom-field-options/1"
        },
        "attributes": {
          "value": "Option 1"
        },
        "relationships": {
          "custom-field": {
            "links": {
              "self": "https://api.teamtailor.com/v1/custom-field-options/1/relationships/custom-field",
              "related": "https://api.teamtailor.com/v1/custom-field-options/1/custom-field"
```

**Review by frysch — CHANGES_REQUESTED** [review]

File: source/includes/custom-field-options/_update.md.erb (hunk 7 above)
Comment (forsner) [reply source/includes/custom-field-options/_update.md.erb:52]: Checked it multiple times! forgot some anyways 🙁

**Review by frysch — APPROVED** [review]
🎉

**Pull request merged**